flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/IcebergTableSource.java

```diff
@@ -49,7 +51,8 @@
 public class IcebergTableSource
     implements ScanTableSource, SupportsProjectionPushDown, SupportsFilterPushDown, SupportsLimitPushDown {
 
-  private int[] projectedFields;
+  private int[][] projectedFields;
+  private int[] topProjectedFields;
```

Comment:
What is this `topProjectedFields` variable meaning ?

Reply:
I just cannot understand the underlying meaning from the variable name.

Reply:
`topProjectedFields` is the index of the root level projection field. For example, if the projection field is `[[5, 1],[5, 2],[3, 1]]`, the top projection field will be `[5, 3]`.